src/prefect/context.py

```diff
@@ -234,9 +233,9 @@ class EngineContext(RunContext):
     flow: Optional["Flow"] = None
     flow_run: Optional[FlowRun] = None
     autonomous_task_run: Optional[TaskRun] = None
-    task_runner: BaseTaskRunner
+    task_runner: Any
```

Comment:
i was slightly over-complicating this in the case where we want to re-use a task runner, we can just pass the task runner instance along instead of entering a null context and then the typing stays the same

updated in [1a5a047](https://github.com/PrefectHQ/prefect/pull/11806/commits/1a5a04746f295843a855c5e29cd280561ef7e831)